src/app/components/student/student.component.ts

```diff
@@ -103,4 +135,61 @@ export class StudentComponent implements OnInit {
       return 'asc';
     }
   }
+
+  filterTable(field: string, event): void {
+    this.filterSubject.next({field: field, value: event.target.value});
+  }
+
+  getFilterArray(newFilter: FilterField, actualFilterArray: Array<FilterField>): Array<FilterField> {
+    let filterArray = [...actualFilterArray];
+    const fieldPredicate = (item: FilterField) => item.field === newFilter.field;
+
+    if (filterArray.length === 0) {
+      if (newFilter.field && newFilter.value) {
+        filterArray.push(newFilter);
+        return filterArray;
+      } else {
+        return filterArray;
+      }
+    } else {
+      const existFilter = filterArray.find(fieldPredicate);
+
+      if (existFilter) {
+        const existIndex = filterArray.findIndex(fieldPredicate);
+
+        if (newFilter.field && newFilter.value) {
+          filterArray.splice(existIndex, 1);
+          filterArray.push(newFilter);
+        } else {
+          filterArray.splice(existIndex, 1);
+        }
+      } else {
+        filterArray.push(newFilter);
+      }
+
+      return filterArray;
+    }
+  }
+
+  setQuery(filterArray: Array<FilterField>): string {
```

Comment:
What exact query is setting here?
And I guess it's actually a get method, cause has a return statement